wpcom-legacy-redirector.php

```diff
@@ -135,9 +154,9 @@ static function get_redirect_uri( $url ) {
 
 				return false;
 			} elseif ( 0 !== $redirect_post->post_parent ) {
-				return get_permalink( $redirect_post->post_parent );
+				return add_query_arg( $protected_param_values, get_permalink( $redirect_post->post_parent ) ); // Add Whitelisted Params to the Redirect URL.
```

Comment:
Hey Philip, not sure what the ask is here on this. We're still in the `get_redirect_uri` function. So basically if there are any query params that are white listed to be passed through at this point they get added to the redirect_uri returned.